streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -28,11 +29,9 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 
-/**
- * Wrapper over StreamThread that implements StateStoreProvider
- */
-public class StreamThreadStateStoreProvider implements StateStoreProvider {
+public class StreamThreadStateStoreProvider {
```

Comment:
With this change, just Global and Wrapping StateStoreProviders exist? IIUC, most of the `xxxStore` classes are just accessing the `Wrapping..` store provider? Makes me wonder, if we should just use the `QueryableStoreProvider` everywhere and cull the interface..

Anyway, I am not familiar enough with this part of the code. So I leave it to you..

Reply:
I also feel that we can collapse the layered interfaces a bit further after we've changed this, since the original motivation of having it is just to "stitch" the global stores and local stores together when exposing as `KafkaStreams#stores`.

We can consider that in a separate, cleanup PR afterwards.

Reply:
No, StateStoreProvider is used in the public API as part of QueryableStoreType. Its role is well defined already. That's the one that we should continue to use "everywhere".

`QueryableStoreProvider` is a utility for selecting either the global store to return OR packaging the thread store providers _into_ a `WrappingStoreProvider` and using it to actually create a usable store. It's not a `StateStoreProvider` at all.

`WrappingStoreProvider` on the other hand _is_ a `StateStoreProvider`, and adapts a `List<StreamThreadStoreProvider>` to the `StateStoreProvider` interface.

`GlobalStoreProvider` is also a `StateStoreProvider`.

So, if anything, we could increase clarity by renaming some of these classes to more accurately reflect their roles, but all the involved classes have well defined and necessary roles right now.

Reply:
@guozhangwang I think we wrote our comments at the same time. Just to respond directly, I do think that we can clean up the code base in a follow-on change (by renaming stuff that's named misleadingly now), but I don't think we can collapse the hierarchy at all.

Reply:
Thanks! SGTM.

Reply:
I can work on the follow-on change.

WrappingStoreProvider -> WrappingStateStoreProvider
GlobalStoreProvider -> GlobalStateStoreProvider

QueryableStoreProvider is left as is.